### ISSUE TITLE
List of Reverb Projects

### ISSUE THREAD

**Issue by SebastianJiroSchlecht** [artificial-audio/pyFDN#123]
Here is a list of possible topics suitable for a short student project:

- Spring reverb - with higher order allpass filters (see Parker)
- Plate reverbs - see Bilbao 
- zita_rev1
- Scattering Delay Networks (SDN)
- Treeverb
- Directional reverbs (see Alary)
- Binaural reverbs
- Interleaved velvet noise
- Common slope reverbs
- Time-variation algorithms
- EQ filter designs
- Allpass filters
- Fade-in FDNs
- Early reflection design
- Modal matching
- Non-linear reverbs
- Flutter Echoes
- Non-diffuse reverbs
- Learning and reverb matching
- Reverberation Enhancement Systems
- ART system (energetic system) to pyFDN
- Frequency-dependent Schroeder allpass (and allpass FDN)
- Nested allpass with short to long and follow allpass in loop idea (#185 )
- Delay Sampling - based on free path distribution (see Kuttruff and [Holmin](https://arxiv.org/pdf/1702.08096) and [Beranke](https://pubs.aip.org/asa/jasa/article/135/1/223/564590/Mean-free-paths-in-concert-and-chamber-music-halls))
- [Griesinger's progenitor reverb](https://www.convexoptimization.com/wikimization/index.php/Dattorro_Convex_Optimization_of_a_Reverberator)
- [Piano Sustain Pedal Effect](http://asa.scitation.org/doi/abs/10.1121/1.2756172?journalCode=jas)
- [1000 years of Reverbs by Sean Costello](https://valhalladsp.wordpress.com/wp-content/uploads/2015/06/aes2015reverbpresentation.pdf) - some later talks have further updates
- Classic Shimmer: Chain of 16 comb allpass (mono and stereo) with one octaver in a single feedback loop
- Matrix based on Butterfly rotation by Gerzon (1972)

See  Fifty Years of Artificial Reverberation for more ideas.

Further examples:
- example_unilossless (possibly new example)
- example_delay_choice (new example) - also implement equilateral approximation
- example Binaural FDN (Menzer and Faller) - IACC matching

New modules:
- TD modules: time-varying delays; time-varying allpass; octaver

Also, analysis/synthesis type of projects are possible:

- Loss functions, spatial, temporal, spectral
- Differentiable Echo density (Mezza)
- RIR datasets (suitable for matching and benchmarking)
- Perceptual evaluation: Metallicness, Naturalness
- Analysis: Decay, Tone, Stability, Allpassness, Echo Density
- Graph to (vanilla) FDN: convert FLAMO graph automatically
- Look at pyrato library to see opportunities for reusing analysis or synthesis code.
- Integrate speech2fdn as a module/example
- Alternative Model Decomposition with PyTorch by applying autograd to the Newton/EAI directly.

Visualizations
- Model Decomposition Visualizer (moving paths and so)
- Plot z-Plane Landscape (related to modal)
- FDN movie remake
 
Apps
- Realtime Interactive App for Common Slopes with many sources and receivers. E.g., 2D floor plan with interactive source and receiver movements.
